x-pack/plugins/security_solution/public/common/components/sourcerer/use_pick_index_patterns.tsx

```diff
@@ -190,9 +237,17 @@ export const usePickIndexPatterns = ({
     [dataViewId, defaultDataViewId, isModified, isOnlyDetectionAlerts, kibanaDataViews]
   );
 
+  useEffect(() => {
+    isHooklAlive.current = true;
```

Comment:
icky

Reply:
yes this hook is constantly getting re-init, so yes it is icky.